### PR TITLE
Cross-platform java detection

### PR DESCRIPTION
**Pull request by maxim-belkin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Adapting java detection code to work on Linux.
Moved current implementation to `extend/os/mac/system_config.rb`.
New default is based on `java -version` and works on both platforms.

CC @sjackman

### CODE REVIEW

**Comment by sjackman** [comment]
Thanks, Maxim!

File: Library/Homebrew/system_config.rb [hunk 1]
```diff
@@ -131,16 +131,10 @@ def kernel
...
-      java_xml = Utils.popen_read("/usr/libexec/java_home", "--xml", "--failfast", err: :close)
+      return "N/A" unless which "java"
+      java_version = Utils.popen_read("java", "-version")
       return "N/A" unless $CHILD_STATUS.success?
-      javas = []
-      REXML::XPath.each(REXML::Document.new(java_xml), "//key[text()='JVMVersion']/following-sibling::string") do |item|
-        javas << item.text
-      end
-      javas.uniq.join(", ")
+      java_version[/java version "([0-9\._]+)"/, 1] || "N/A"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/system_config.rb]: `0-9` can be `\d`

Comment (maxim-belkin) [reply Library/Homebrew/system_config.rb]: done!

**Comment by MikeMcQuaid** [comment]
Thanks again @maxim-belkin!

**Comment by sjackman** [comment]
Thanks for merging, Mike!

**Pull request merged**